src/windows/system.rs

```diff
@@ -526,6 +546,37 @@ fn get_reg_string_value(hkey: HKEY, path: &str, field_name: &str) -> Option<Stri
     Some(s)
 }
 
+fn get_reg_value_u32(hkey: HKEY, path: &str, field_name: &str) -> Option<[u8; 4]> {
+    let c_path = utf16_str(path);
+    let c_field_name = utf16_str(field_name);
+
+    let mut new_hkey: HKEY = std::ptr::null_mut();
+    if unsafe { RegOpenKeyExW(hkey, c_path.as_ptr(), 0, KEY_READ, &mut new_hkey) } != 0 {
+        return None;
+    }
+
+    let mut buf_len: DWORD = 4;
+    let mut buf_type: DWORD = 0;
+    let mut buf = [0u8; 4];
+    loop {
```

Comment:
I'm not sure to understand what this loop is doing... If `0`, you break, otherwise you return `None`. Should a `if` be enough?

Reply:
Fixed in probably the best commit number I've ever had bb8ccdd